apkleaks/apkleaks.py

```diff
@@ -88,28 +91,43 @@ def integrity(self):
 
 	def decompile(self):
 		util.writeln("** Decompiling APK...", col.OKBLUE)
-		args = [self.jadx, self.file, "-d", self.tempdir]
+
+		#If the path contains a space..
+		#This is the case if apkleaks was installed on windows through PyPi, as apkleaks.exe ends up in the path "C:/Program Files/Python39/Lib/site-packages/jadx/bin/jadx.bat" (in this example, under python39)
+		#On windows, if an executable's path doesn't contain spaces but is it quoted anyway, the command fails. We must only quote the path if it has spaces.
+		if(self.jadx.__contains__(" ")):
+			args = ["\"" + self.jadx + "\"", self.file, "-d", self.tempdir]
+		else:
+			args = [self.jadx, self.file, "-d", self.tempdir]
+		
+
 		try:
 			args.extend(re.split(r"\s|=", self.disarg))
 		except Exception:
 			pass
+
+		#add single quotes to every parameter
 		comm = "%s" % (" ".join(quote(arg) for arg in args))
+
+		#jadix.bat doesn't like single quotes. Replace them with double quotes
 		comm = comm.replace("\'","\"")
+
+		#If the path to jadx contains a space
+		if(self.jadx.__contains__(" ")):
+			#eliminate duplicate quotes
+			comm = comm.replace("\"\"","\"")
+		
 		os.system(comm)
 
 	def extract(self, name, matches):
 		if len(matches):
-			stdout = ("[%s]" % (name))
-			util.writeln("\n" + stdout, col.OKGREEN)
-			self.fileout.write("%s" % (stdout + "\n" if self.json is False else ""))
 			for secret in matches:
 				if name == "LinkFinder":
 					if re.match(r"^.(L[a-z]|application|audio|fonts|image|kotlin|layout|multipart|plain|text|video).*\/.+", secret) is not None:
 						continue
 					secret = secret[len("'"):-len("'")]
-				stdout = ("- %s" % (secret))
-				print(stdout)
-				self.fileout.write("%s" % (stdout + "\n" if self.json is False else ""))
+				print(name + ": " + secret)
+				self.fileout.write("%s" % (name + ": " + secret + "\n" if self.json is False else ""))
```

Comment:
Revert this.

Reply:
Having grep-able output is a big plus IMO. It allows automation and an overall faster workflow